pytorch_to_returnn/naming/tensor.py

```diff
@@ -104,7 +104,11 @@ def get_returnn_axis_description(self, torch_axis: int) -> str:
     dim_tag = self.returnn_data.get_dim_tag(axis)
     assert dim_tag.kind == DimensionTag.Types.Spatial
     if dim_tag.dyn_size is not None:
-      return f"stag:{dim_tag.description}"
+      axes_with_same_dim_tag = [ax for ax in range(ndim) if dim_tag == self.returnn_data.get_dim_tag(ax)]
+      if len(axes_with_same_dim_tag) == 1:
+        return f"stag:{dim_tag.description}"
+      else:
+        return f"stag-single:{axis - ndim}:{dim_tag.description}"
```

Comment:
The index calculation looks wrong.
```suggestion
        return f"stag-single:{axes_with_same_dim_tag.index(axis) - len(axes_with_same_dim_tag)}:{dim_tag.description}"
```